File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/services/ledger/validations/SplitSideValueValidator.java
package com.decrux.pocketr.api.services.ledger.validations;

import com.decrux.pocketr.api.entities.db.ledger.SplitSide;
import com.decrux.pocketr.api.entities.dtos.CreateSplitDto;
import com.decrux.pocketr.api.exceptions.BadRequestException;
import java.util.List;
import org.springframework.stereotype.Component;

@Component
public class SplitSideValueValidator {

    public void validate(List<CreateSplitDto> splits) {
        for (CreateSplitDto split : splits) {
            try {
                SplitSide.valueOf(split.getSide());
            } catch (IllegalArgumentException ignored) {
                throw new BadRequestException("Invalid split side: " + split.getSide());
            }
        }
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/services/ledger/validations/SplitSideValueValidator.kt
package com.decrux.pocketr.api.services.ledger.validations

import com.decrux.pocketr.api.entities.db.ledger.SplitSide
import com.decrux.pocketr.api.entities.dtos.CreateSplitDto
import com.decrux.pocketr.api.exceptions.BadRequestException
import org.springframework.stereotype.Component

@Component
class SplitSideValueValidator {
    fun validate(splits: List<CreateSplitDto>) {
        splits.forEach { split ->
            try {
                SplitSide.valueOf(split.side)
            } catch (_: IllegalArgumentException) {
                throw BadRequestException("Invalid split side: ${split.side}")
            }
        }
    }
}
